sdk/resourcemanager/Azure.ResourceManager.Core/src/Resources/SubResourceReadOnly.cs

```diff
@@ -7,12 +7,19 @@ namespace Azure.ResourceManager.Core
     /// A class representing a sub-resource that contains only the read-only ID.
     /// </summary>
     [ReferenceType]
-    public abstract class SubResourceReadOnly
+    public partial class SubResourceReadOnly
     {
+        /// <summary> Initializes a new instance of SubResourceReadOnly. </summary>
+        /// <param name="id"> ARM resource Id. </param>
+        internal SubResourceReadOnly(string id)
```

Comment:
needs to be protected internal otherwise the classes that inherit this from the other RPs won't be able to access it

Reply:
Absolutely!